src/relations/db.py

```diff
@@ -135,7 +134,7 @@ def _on_relation_joined(self, join_event: RelationJoinedEvent):
         If the backend relation is fully initialised and available, we generate the proposed
         database and create a user on the postgres charm, and add preliminary data to the databag.
         """
-        if not self.charm.backend_postgres:
+        if not self.charm.backend.postgres:
```

Comment:
might you need to check if PBG is also ready and running?

Reply:
PGB doesn't actually have to be running for this relation to come online, but it does need to have a valid config file, so I've added a check for that.